iq_learn/train_iq.py

```diff
@@ -179,22 +180,36 @@ def main(cfg: DictConfig):
             expert_batch = expert_memory_replay.get_samples(
                 agent.batch_size, agent.device
             )
-            expert_obs, _, expert_action, __, ___, expert_cond, true_traj_idx = expert_batch
+            expert_obs, _, expert_action, __, ___, expert_cond_detached, true_traj_idx = expert_batch
+            unique_idx = []
+            expert_obs_unique = []
+            expert_action_unique = []
+            for i, traj_idx in enumerate(true_traj_idx):
+                if traj_idx not in unique_idx:
+                    unique_idx.append(traj_idx)
+                    expert_obs_unique.append(expert_obs[i])
+                    expert_action_unique.append(expert_action[i])
+            expert_obs_unique = torch.stack(expert_obs_unique, dim=0)   
```

Comment:
we want conditions to be unique, not obs & act pairs. 

concrete example: 
- expert batch of size 32
- you can have 32 obs-act pairs
- you can have 4 unique conditions corresponding to these pairs

do you think does this make sense

Reply:
Changed to the following:
- expert batch of size 32
- 32 obs-act pairs
- 32 conditions corresponding to these pairs